cukinia

```diff
@@ -891,6 +897,66 @@ _cukinia_knoerror()
 	return $ret
 }
 
+# _cukinia_network_interface_up: check if a network interface is up
+# arg1: interface name
+_cukinia_network_interface_up() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" is ${__not:+NOT }up"
+
+	ip link show "$interface" up | grep -q 'state UP' >/dev/null 2>&1 && return 0 || return 1
+}
+
+# _cukinia_network_interface_ipv4: check if an interface has an IPv4 address
+# arg1: interface name
+_cukinia_network_interface_ipv4() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" has ${__not:+NO }IPv4 address"
+
+	ip -4 addr show dev "$interface" | grep -q inet && return 0 || return 1
+}
+
+# _cukinia_network_interface_global_ipv6: check if a network interface has a global IPv6 address
+# arg1: interface name
+_cukinia_network_interface_global_ipv6() {
+	local interface="$1"
+
+	_cukinia_prepare "Checking if interface \"$interface\" has ${__not:+NO }global IPv6 address"
+
+	ip -o -6 address show dev "$interface" scope global | grep -q "inet6" && return 0 || return 1
```

Comment:
`-o` seems like an option that should be everywhere: `output each record on a single line, replacing line feeds with the '\' character. This is convenient when you want to count records with wc(1) or to grep(1) the output.`

Reply:
Done